### ISSUE TITLE
Inputs for customization & usability improvements

### ISSUE THREAD

**Issue by speedbiker** [openstreetmap/iD#4281]
After using iD-Editor now for more than 2 years I would like to point of some improvements in usablity and the possibilty of customization of buttons/fields.

Now that I have worked extensively with iD-editor for more than 2 years (about 1,500,000 changes to the map) and I would like to make some suggestions to improve/optimize the user interface as well as the usability:

1. For the daily work a favorite toolbar (or similar) would be very pleasant, in which I can insert up to 10 regularly used buttons (for example, surfaces, nodes, lines), which remain permanently/very often (user-specific). There are so many tags in the meanwhile in OSM, that it's a lot of work to search again and again for a field.
Currently, just the last 4 or 5 used buttons/field are saved, but not permanently.
3. A keyboard command which assigns the last command to the object that has just been drawn. For example If I draw a surface/aera and I have also drawn a surface that has the tag "building = yes" one step before, the iD editor should automatically assign this tag when pressing a simple keyboard combination (max. up to 2 buttons).
4. Show/hide fields: Especially when inputting addresses, it would be very helpful to be able to show/hide parts of the address. The default address field in the iD editor was as follows:

- addr: street
- addr: housenumber
- addr: postalCode
- addr: city

Unfortunately, the "addr: unit" field, which, for example, In Austria is meaningless for addressing. It costs however space on the screen, which is as more important the smaller the screens become (keyword tablets, smartphones). A user-specific adaptation of usability in the incorporation of addresses would result in an enormous increase of working speed and efficiency. Because, those fields, which I do not need as a Mapper in my region (e.g. the field "addr: unit") I simply hide. But on the other hand, the field "Addr: place", which is very common in my geographic region, iD editor should show. It is important that you can also jump from field to field with the tab key as is now. Conversely, I can leave fields, which are important to me, or simply replace them in the template. In my case, the "Addr: unit" with "Addr: place".
Important: **these settings should be stored** with the logged in user, preferably on the server, otherwise by cookies.
In general, from my point of view any mapper has other demands and needs. Some are mapping only landscape, others just buildings and others just POI's. A customization helps mappers who are working intensively by increasing speed/efficiency on the one side but also helps to manage a better overview of currently done work.
5. When drawing surfaces, there should be the option to draw right-angled surfaces, which would be very helpful especially for house mapping (which can be switched by keyboard command). From the process I think about this: usual selection of a surface > keyboard command (only 1 key) and a rectangle of this area is created automatically. Quite similar to the JOSM editor.

**Comment by bhousel** [comment]
Hey @speedbiker thanks for the detailed report!  I think all of your suggestions are covered by existing issues on our issue tracker.. See below:

> 1. For the daily work a favorite toolbar (or similar) would be very pleasant, in which I can insert up to 10 regularly used buttons

#4125 / #590 / #839 

> 2. A keyboard command which assigns the last command to the object that has just been drawn.

also #590

> 3. Unfortunately, the "addr: unit" field, which, for example, In Austria is meaningless for addressing. It costs however space on the screen..

already done in #4301 

> 4. Important: these settings should be stored with the logged in user, preferably on the server, otherwise by cookies.

#3002

> 5. When drawing surfaces, there should be the option to draw right-angled surfaces, 

#528 / #2699

Some of this stuff is blocked by #3123 because I don't want to clutter the iD UI by adding a bunch of buttons to the top bar for more toggle-able modes.
